## Preprint published article detail for Biorxive

In [10]:
import numpy as np
import pandas as pd
import requests
import json

In [96]:
start_date = '2023-01-01'
end_date = '2023-01-31'
cursor = 0
response = requests.get('https://api.biorxiv.org/pubs/biorxiv/' + start_date +'/' + end_date + '/' + str(cursor))

if response.status_code == 200:
    results = response.json()
else:
    print("Request failed: ", response.status_code)

In [97]:
print(results["messages"])
# print(results["collection"][0])
print(response.url)
# print(results)

[{'status': 'ok', 'interval': '2023-01-01:2023-01-31', 'cursor': 0, 'count': 100, 'total': 1260}]
https://api.biorxiv.org/pubs/biorxiv/2023-01-01/2023-01-31/0


In [98]:
# This array hold the data received from API this way its easy to add data as it retrive data from API as 100 chunks
# collection_data.append here will add the first 100 results to the collection_data list
collection_data = []
collection_data.append(results)
# collection_data

Single API call retrive only 100 data points. Cursor can specify from which point the next 100 data point start
Therefore get the total data points and divided from 100 to get how many iterations the API call should run

In [99]:
# To get the how many times the API should run = Total data / 100 
count = results['messages'][0]['total'] / 100
count = round(count)
count

13

In [100]:
# Below code will add the data to collection_data from 100th cursor since the first 100 already added to the collection_data
temp_cursor = 0

for i in range(count):
    temp_cursor = temp_cursor + 100
    print(f"{temp_cursor} started")
    response = requests.get('https://api.biorxiv.org/pubs/biorxiv/' + start_date +'/' + end_date + '/' + str(temp_cursor))

    if response.status_code == 200:
        results = response.json()
        collection_data.append(results)
        print(f"Cursor {temp_cursor} added to the collection_data")
    else:
        print("Request failed: ", response.status_code)


100 started
Cursor 100 added to the collection_data
200 started
Cursor 200 added to the collection_data
300 started
Cursor 300 added to the collection_data
400 started
Cursor 400 added to the collection_data
500 started
Cursor 500 added to the collection_data
600 started
Cursor 600 added to the collection_data
700 started
Cursor 700 added to the collection_data
800 started
Cursor 800 added to the collection_data
900 started
Cursor 900 added to the collection_data
1000 started
Cursor 1000 added to the collection_data
1100 started
Cursor 1100 added to the collection_data
1200 started
Cursor 1200 added to the collection_data
1300 started
Cursor 1300 added to the collection_data


In [101]:
collection_data_list = []

for item in collection_data:
    for sub_item in item['collection']:
        collection_data_list.append(sub_item)

In [102]:
len(collection_data_list)

1260

In [103]:
# df23 = pd.DataFrame(collection_data_list)
# df23.to_csv(r'C:\Users\Hiras\Analysis_Hiras\biorxive\output\brxive_df23.csv')

From 2020-01-01 : 2020-12-31 total number of results 22074

From 2021-01-01 : 2021-12-31 total number of results 25114

From 2022-01-01 : 2022-12-31 total number of results 19742

From 2023-01-01 : 2023-01-31 total number of results 1260